app/models/manageiq/providers/lenovo/physical_infra_manager/event_catcher/stream.rb

```diff
@@ -44,29 +47,45 @@ def parse_events(events)
     end
 
     # Update the @last_event_ems_ref with the new last ems_ref if to exist new events
-    @last_event_ems_ref = parsed_events.last[:ems_ref]
+    @last_event_ems_ref = parsed_events.last[:ems_ref].to_i
     parsed_events
   end
 
   def events
     expression = { :filterType => 'FIELDNOTREGEXAND', :fields => filter_fields }
-
     opts = {'filterWith' => expression.to_json}
 
+    @last_cn ||= get_last_cn(opts)
+    opts['headers'] = pagination unless @last_event_ems_ref >= @last_cn
```

Comment:
__Suggestion__
I think that the positive condition would be better understood. In this case:
`opts['headers'] = pagination if @last_event_ems_ref < @last_cn`